08/kth-missing-positive-number.checkpoint.md

```diff
@@ -208,6 +208,106 @@ Code solution to a subproblem
 
 <!-- ======================= END CHALLENGE ======================= -->
 
+<!-- >>>>>>>>>>>>>>>>>>>>>> BEGIN CHALLENGE >>>>>>>>>>>>>>>>>>>>>> -->
+<!-- Replace everything in square brackets [] and remove brackets  -->
+
+### !challenge
+
+* type: code-snippet
+* language: python3.6
+* id: 5a380194-96e2-4b87-9aa7-2fe407448e1f
+* title: Coding Solution
+* points: 3
+* topics: python, pse, python-lists
+
+##### !question
+
+Write the code to solve the problem here.  You can use the tests to guide your solution.
+
+##### !end-question
+
+##### !placeholder
+
+```python
+def kth_missing_positive_number(numbers, k):
+    '''
+    INPUT: List of numbers in increating order & a positive integer k
+    OUTPUT: The kth missing number
+    '''
+
+    pass
+```
+
+##### !end-placeholder
+
+##### !tests
+
+```py
+import unittest
+import main as p
+import numpy as np
+
+class TestPython1(unittest.TestCase):
+  def test_example_one(self):
+    # Arrange
+    list = [2,3,4,7,11]
+    k = 5
+
+    # Act
+    answer = p.kth_missing_positive_number(list, k)
+
+    # Assert
+    self.assertEqual(9, answer)
+
+  def test_example_2(self):
+
+    # Arrange
+    list = [1,2,3,4]
+    k = 2
+
+    # Act
+    answer = p.kth_missing_positive_number(list, k)
+
+    # Assert
+    self.assertEqual(6, answer)
+
+  def test_three(self):
+
+    # Arrange
+    list = [2,3,4,7,11]
+    k = 5
+
+    # Act
+    answer = p.kth_missing_positive_number(list, k)
+
+    # Assert
+    self.assertEqual(9, answer)
+
+  def test_k_is_one_and_one_is_missing(self):
+
+    # Arrange
+    list = [2,3,4,7,11]
+    k = 1
+
+    # Act
+    answer = p.kth_missing_positive_number(list, k)
+
+    # Assert
+    self.assertEqual(1, answer)
```

Comment:
Because we import `main as p`, we have to use `p` to access our methods, as described below.

However, if an error comes up, this will confuse the students. The stack trace will them that there's an error about `p.kth_missing_positive_number`, and they'll ask "What is `p`?"

If we import `main as *`, we can circumvent this:

```suggestion
import main as *

class TestPython1(unittest.TestCase):
  def test_example_one(self):
    # Arrange
    list = [2,3,4,7,11]
    k = 5

    # Act
    answer = kth_missing_positive_number(list, k)

    # Assert
    self.assertEqual(9, answer)

  def test_example_2(self):

    # Arrange
    list = [1,2,3,4]
    k = 2

    # Act
    answer = kth_missing_positive_number(list, k)

    # Assert
    self.assertEqual(6, answer)

  def test_three(self):

    # Arrange
    list = [2,3,4,7,11]
    k = 5

    # Act
    answer = kth_missing_positive_number(list, k)

    # Assert
    self.assertEqual(9, answer)

  def test_k_is_one_and_one_is_missing(self):

    # Arrange
    list = [2,3,4,7,11]
    k = 1

    # Act
    answer = kth_missing_positive_number(list, k)

    # Assert
    self.assertEqual(1, answer)
```

Also we aren't using numpy here so we can drop the import

I haven't tested this myself with a solution